### PR TITLE
feat(projectman): add a new resource to manage project

### PR DESCRIPTION
**Pull request by hmbig2**
**What this PR does / why we need it**:
add a new resource to manage project

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [X] Tests added/passed.
* [X] Documentation updated.
* [X] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/projectman' TESTARGS='-run=TestAccProject_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/projectman -v -run=TestAccProject_basic -timeout 360m -parallel 4
=== RUN   TestAccProject_basic
=== PAUSE TestAccProject_basic
=== CONT  TestAccProject_basic
--- PASS: TestAccProject_basic (49.24s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/projectman        49.302s
```

### CODE REVIEW

File: huaweicloud/services/acceptance/projectman/resource_huaweicloud_projectman_project_test.go [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils"
+)
+
+func getProjectResourceFunc(config *config.Config, state *terraform.ResourceState) (interface{}, error) {
+	var (
+		getProjectHttpMethod = strings.ToUpper("GET")
```
Comment (niuzhenguo) [inline huaweicloud/services/acceptance/projectman/resource_huaweicloud_projectman_project_test.go]: This seems a bit werid, can we get rid of calling ToUpper to a fixed string.

Comment (hmbig2) [reply huaweicloud/services/acceptance/projectman/resource_huaweicloud_projectman_project_test.go]: fix

File: huaweicloud/config/endpoints.go [hunk 2]
```diff
@@ -614,6 +614,12 @@ var allServiceCatalog = map[string]ServiceCatalog{
 		Version: "v1",
 		Product: "VOD",
 	},
+
```
Comment (Lance52259) [inline huaweicloud/config/endpoints.go]: missing category comment: `catalog for Developer Services`

File: docs/resources/projectman_project.md [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+* `archive` - Whether the project is archived.
+
+* `project_code` - The project code.
+
+* `project_num_id` - The number id of project.
+
+## Import
+
+The Project can be imported using the `id`, e.g.
```
Comment (Lance52259) [inline docs/resources/projectman_project.md]: s/Project/project

File: docs/resources/projectman_project.md [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+  Changing this parameter will create a new resource.
+
+* `template_id` - (Optional, Int, ForceNew) The template id which used to create project.
+  Changing this parameter will create a new resource.
+
+## Attributes Reference
+
+In addition to all arguments above, the following attributes are exported:
+
+* `id` - The resource ID.
```
Comment (Lance52259) [inline docs/resources/projectman_project.md:52]: maybe `The {lowercase(product)} ID.` is better.

File: docs/resources/projectman_project.md [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+* `name` - (Required, String) The project name.  
+  The name can contain 1 to 128 characters.
+
+* `type` - (Required, String, ForceNew) The type of project.  
+  The valid values are **scrum**, **xboard**, **basic**, **phoenix**.
+  Changing this parameter will create a new resource.
+
+* `description` - (Optional, String) The description about the project.
+
+* `enterprise_project_id` - (Optional, String, ForceNew) The enterprise project id of the project.
```
Comment (Lance52259) [inline docs/resources/projectman_project.md]: s/id/ID

File: huaweicloud/config/endpoints.go (hunk 2 above)
Comment (hmbig2) [reply huaweicloud/config/endpoints.go]: added

File: docs/resources/projectman_project.md (hunk 5 above)
Comment (hmbig2) [reply docs/resources/projectman_project.md]: ok

File: docs/resources/projectman_project.md (hunk 3 above)
Comment (hmbig2) [reply docs/resources/projectman_project.md]: OK

File: docs/resources/projectman_project.md (hunk 4 above)
Comment (hmbig2) [reply docs/resources/projectman_project.md:52]: no, keep it as it is.

File: huaweicloud/config/endpoints.go [hunk 6]
```diff
@@ -614,6 +614,13 @@ var allServiceCatalog = map[string]ServiceCatalog{
...
+	// catalog for DevCloud
+	"projectman": {
+		Name:    "projectman-ext",
+		Version: "v4",
```
Comment (Lance52259) [inline huaweicloud/config/endpoints.go:633]: The `version` is unnecessary.

**Pull request merged**